doc/paper/paper.tex

```diff
@@ -0,0 +1,1012 @@
+    can operate directly against S3. This isolates heavy and bursty ML-related read
+    workloads from the comparatively lighter-weight streaming playback and
+    summarization workloads supported by the database server, and reduces the cost
+    and complexity of infrastructure management by eliminating the need to scale a
+    database deployment in step with the job workloads.
+
+    Operational burden for the data infrastructure admin is minimized by offloading
+    primary storage and index structure to S3 and allowing the admin to use S3
+    tooling (lifecycle policies, retention policies, etc) to govern the garbage
+    collection of the tree. To keep users from being exposed to deleted objects,
+    the admin must ensure the tree storage is always truncated ahead of the
+    physical deletion schedule. Since dp3 supports any S3-compatible storage, it
+    can be deployed in on-premise environments in situations where cloud offload
+    bandwidth is limited at the point of ingestion.
+
+    dp3 is initially targeting a smooth integration experience with the ROS
+    \cite{Quigley09} ecosystem and the MCAP \cite{MCAP} format, to become a viable
+    \q{plug and play} infrastructure primitive for ROS and MCAP users. To
+    facilitate this, dp3 accepts MCAP files as its input format, stores data in
+    MCAP format in the leaves of the tree, and streams MCAP out during playback.
+    The messages recorded by the user are repackaged and compressed, but not
+    otherwise transformed from their original encoding, making dp3’s output
+    compatible with any internal log tooling the user has developed. Once dp3’s
+    supported set of message encodings is expanded, it should be fully usable as a
+    storage component by any ROS user.
+
+    The first section of this paper will elaborate on some of the design goals and
+    high-level context surrounding the project. The second section will describe
+    the implementation of key components in greater detail.
+
+    \section{Background}
+    \subsection{Multimodal log management}
+    In robotics, it is common to use the word \q{logging} to refer to the process of
+    writing multimodal diagnostic messages to network or disk for followup
+    analysis. Perhaps by intent, the term serves to emphasize the parallels with
+    traditional server application logging, i.e applications log text, robots log
+    point clouds.
+
+    Server application log management platforms are plentiful and follow
+    well-established patterns. Some examples of these include Sumo logic
+    \cite{sumologic}, Datadog logging \cite{datadoglog}, and the log management
+    interfaces of the three major cloud providers: Google Cloud Logging \cite{gcplog}, AWS
+    Cloudwatch Logs \cite{cloudwatch}, and Azure Monitor Logs \cite{azlog}.
+
+    From the user’s point of view these log management solutions are mostly
+    interchangeable. The core interface consists of, (1) Search controls:
```

Comment:
Since items in the list are periodized, we should format this as a list? Or use a comma as a delimiter between items?

Reply:
I've gone back and forth on the formatting of the lists in the paper, and settled on inline here because I found it most attractive to look at. I will change the separators to commas or semicolons.